interfaces/input_parameters/ring_section.py

```diff
@@ -1,77 +1,260 @@
+        Synchronous total energy program for each segment of the ring
+        :math:`E_{s,k,n} = \sqrt{ p_{s,k,n}^2 + m^2 }` [eV]
+    kin_energy : float matrix [n_sections, n_turns+1]
+        Synchronous kinetic energy program for each segment of the ring
+        :math:`E_{s,kin} = \sqrt{ p_{s,k,n}^2 + m^2 } - m` [eV]
+    delta_E : float matrix [n_sections, n_turns]
+        Gain in synchronous total energy from one point to another,
+        for all sections,
+        :math:`: \quad E_{s,k,n+1}- E_{s,k,n}` [eV]
+    cycle_time : float array [n_turns+1]
+        Cumulative cycle time, turn by turn, :math:`t_n = \sum_n T_{0,n}` [s].
+        Possibility to extract cycle parameters at these moments using
+        'parameters_at_time'.
+    alpha_order : int
+        Highest order of momentum compaction (as defined by the input). Can
+        be 0,1,2.
+
+    Examples
+    --------
+    >>> # To declare a single-section synchrotron at constant energy:
+    >>> # Particle type Proton
+    >>> from beam.beam import Proton
+    >>> from input_parameters.ring import Ring
+    >>>
+    >>> n_turns = 10
+    >>> C = 26659
+    >>> alpha_0 = 3.21e-4
+    >>> momentum = 450e9
+    >>> ring = Ring(C, alpha_0, momentum, Proton(), n_turns)
 
-        self.section_length = section_length
+    """
 
-        syncDataTypes = ('momentum', 'kin_energy', 'energy', 
-                         'B_field')
+    def __init__(self, section_length, alpha_0,
+                 momentum=None, kin_energy=None, energy=None,
+                 bending_field=None, bending_radius=None,
+                 alpha_1=None, alpha_2=None, **kwargs):
+
+        # Setting section length
+        self.section_length = dTypes._ring_function(section_length)
+
+        # Checking that at least one synchronous data input is passed
+        syncDataTypes = ('momentum', 'kin_energy', 'energy', 'B_field')
         syncDataInput = (momentum, kin_energy, energy, bending_field)
-        assrt.single_not_none(*syncDataInput, msg = 'Exactly one of '
-                              + str(syncDataTypes) + ' must be declared',
-                              exception = excpt.InputError)
+        assrt.single_not_none(*syncDataInput,
+                              msg='Exactly one of '+str(syncDataTypes) +
+                              ' must be declared',
+                              exception=excpt.InputError)
 
+        # Checking that the bending_radius is passed with the bending_field
+        # and setting the bending_radius if defined
+        self.bending_radius = None
         if bending_field is not None and bending_radius is None:
-            raise excpt.InputError("If bending_field is used, bending_radius "
-                                   + "must be defined.")
+            raise excpt.InputError("If bending_field is used, " +
+                                   "bending_radius must be defined.")
+        else:
+            self.bending_radius = dTypes._ring_function(bending_radius)
```

Comment:
Unless bending_radius is expected to change with time I don't see a need for it to be anything other than a float

Reply:
Removed bending_radius as program, back to float

Reply:
I'm going to start experimenting with the circumference as a program soon, so we'll have to revisit this at a later point.